Chapter6/super_process.erl

```diff
@@ -0,0 +1,122 @@
+-module(super_process).
+-export([start_link/2, stop/1, add_child/2, stop_child/2]).
+-export([init/0]).
+
+% Interface
+start_link(Name, ChildSpecList) ->
+	register(Name, spawn_link(super_process, init, [])),
+	start_children(Name, ChildSpecList).
+
+stop(Name) ->
+	Name ! {stop, self()},
+	receive {reply, Reply} -> Reply end.
+
+
+% Exercise 6-3: A Supervisor Process (start_child & stop_child)
+% add_child(Name, {Module, Function, Arguments, Type})
+add_child(Name, Process) ->
+	Name ! {add, self(), Process},
+	receive {reply, Reply} -> Reply end.
+
+stop_child(Name, Id) ->
+	Name ! {remove, self(), Id},
+	receive {reply, Reply} -> Reply end.
+% Interface
+
+
+
+init() ->
+	process_flag(trap_exit, true),
+	loop([]).
+
+start_children(_, []) -> [];
+start_children(Name, [ {Module, Function, _, _} = NewProcess | ChildSpecList]) ->
+	Id = add_child(Name, NewProcess),
+	[{Module,Function,Id} | start_children(Name, ChildSpecList)].
+
+
+% Don't know how to make the module not available to test 6-3 part two (spawn a child whose module is not available?)
```

Comment:
It was updated and should be working as intended, I just forgot to remove the comment